src/Raven.Server/ServerWide/Maintenance/ClusterObserver.cs

```diff
@@ -252,18 +255,39 @@ class ClusterObserver : IDisposable
                     // We are in passive mode and were kicked out of the cluster.
                     return false;
                 }
-                if (previousClusterStats.TryGetValue(mentorNode, out var mentorPrevClusterStats) == false ||
+
+                if (ShouldReassign(topology, promotable, dbName, current, previous))
+                {
+                    var node = FindLeastDbNode(promotable, dbName, current);
+                    if (node == null)
+                    {
+                        if (_logger.IsOperationsEnabled)
+                        {
+                            _logger.Operations($"Somthing is wrong with {dbName} on {promotable}, but we were unable to reassign it.");
+                        }
+                        continue;
+                    }
+                    if (_logger.IsOperationsEnabled)
+                    {
+                        _logger.Operations($"Somthing is wrong with {dbName} on {promotable}, so we reassign it to {node}.");
```

Comment:
Mystery keep the ops teams on their toes. There is something wrong, but we wont tell you what.
Figure it out on your own.